cmd/argoexec/commands/emissary.go

```diff
@@ -39,14 +39,21 @@ func NewEmissaryCommand() *cobra.Command {
 			exitCode := 64
 
 			defer func() {
-				err := ioutil.WriteFile(varRunArgo+"/ctr/"+containerName+"/exitcode", []byte(strconv.Itoa(exitCode)), 0o600)
+				err := ioutil.WriteFile(varRunArgo+"/ctr/"+containerName+"/exitcode", []byte(strconv.Itoa(exitCode)), 0o644)
 				if err != nil {
 					logger.Error(fmt.Errorf("failed to write exit code: %w", err))
 				}
 			}()
 
-			// this also indicates we've started
-			if err := os.MkdirAll(varRunArgo+"/ctr/"+containerName, 0o700); err != nil {
+			// default umask can be 022
+			// setting umask as 0 allow granting write access to other non-root users
+			syscall.Umask(0)
```

Comment:
Should I put this into os_specific?

Reply:
`syscall.Umask(0)` is not available on Windows. So you'll need to move this command into `os_specific`. The Windows implementation can be a no-op.